src/components/structures/RoomView.js

```diff
@@ -1339,18 +1445,52 @@ module.exports = React.createClass({
     },
 
     scrollToBottom: function() {
-        var messagePanel = this.refs.messagePanel;
-        if (!messagePanel) return;
-        messagePanel.scrollToBottom();
+        // if we can't forward-paginate the existing timeline, then there
+        // is no point reloading it - just jump straight to the bottom.
+        //
+        // Otherwise, reload the timeline rather than trying to paginate
+        // through all of space-time.
+        if (this._timelineWindow.canPaginate(EventTimeline.FORWARDS)) {
+            this._loadTimeline();
+        } else {
+            if (this.refs.messagePanel) {
+                this.refs.messagePanel.scrollToBottom();
+            }
+        }
     },
 
     // get the current scroll position of the room, so that it can be
-    // restored when we switch back to it
+    // restored when we switch back to it.
+    //
+    // This returns an object with the following properties:
+    //
+    //    focussedEvent: the ID of the 'focussed' event. Typically this is the
+    //        last event fully visible in the viewport, though if we have done
+    //        an explicit scroll to an explicit event, it will be that event.
+    //
+    //    pixelOffset: the number of pixels the window is scrolled down from
+    //        the focussedEvent.
+    //
+    // If there are no visible events, returns null.
+    //
     getScrollState: function() {
         var messagePanel = this.refs.messagePanel;
         if (!messagePanel) return null;
 
-        return messagePanel.getScrollState();
+        var scrollState = messagePanel.getScrollState();
+
+        if (scrollState.atBottom) {
+            // we don't really expect to be in this state, but it will
+            // occasionally happen when we are in a transition. Treat it the
+            // same as having no saved state (which will cause us to scroll to
+            // last unread on reload).
+            return null;
```

Comment:
You don't really say why you don't expect us to be in this state. A simple `resetScrollState()` would do it...